### ISSUE TITLE
Typedoc generates objects instead of arrays for get/set Signature

### ISSUE THREAD

**Issue by ovasquez** [TypeStrong/typedoc#707]
When generating in `json` format the `getSignature` and `setSignature` of **properties** are being generated as objects and not as arrays.

For some reason the test for the properties (`getter-setter.ts`) work with `npm test`, but when executing the same test from the console the signatures for get/set are generated incorrectly.

**Note:** I'm reporting this as an error because the interface [`DeclarationReflectionContainer<T>`](https://github.com/TypeStrong/typedoc/blob/master/src/lib/serialization/browser.ts#L92) has the **index/get/set** signatures declared as an array.

### Debug results
This can be fixed in the file `src\lib\serialization\serializers\reflections\abstract.ts` although I'm not sure about the full implications of this change.

```
switch (property) {
        case TraverseProperty.GetSignature: // Remove this line
        case TraverseProperty.SetSignature: // Remove this line
        case TraverseProperty.IndexSignature:
          obj[name] = this.owner.toObject(child);
          break;
        default:
          if (!obj[name]) {
            obj[name] = [];
          }
          obj[name].push(this.owner.toObject(child));
          break;
      }
```

### Steps to reproduce
- Copy the file `src/tests/convert/getter-setter/getter-setter.ts` to a new temporary directory.
- Run typedoc to generate `json` for the temp directory.
  `typedoc --json docs.json temp --target es5`

### Excepted Behavior
The `getSignature` and `setSignature` of properties are generated as arrays.

```
"getSignature": [
    {
        "id": 5,
        "name": "__get",
        "kind": 524288,
        "kindString": "Get signature",
        "flags": {},
        "type": {
            "type": "intrinsic",
            "name": "string"
        }
    }
],
```

### Actual Behavior
The `getSignature` and `setSignature` of properties are being generated as objects and not as arrays.

``` 
"getSignature": {
	"id": 5,
	"name": "__get",
	"kind": 524288,
	"kindString": "Get signature",
	"flags": {},
	"type": {
		"type": "intrinsic",
		"name": "string"
	}
},
```

### System details
- OS: Windows 10 64-bit (eng)
- `$ node -v` v6.11.1
- `$ npm -v` 5.6.0
- `$ typedoc -v` TypeDoc 0.10.0 Using TypeScript 2.7.1 from...

**Comment by ovasquez** [comment]
Hi @aciccarello, I apologize for mentioning you in the issue but it seems that you are the only one active in the support/reviewer role.

Do you think that is feasible to accept a pull request with the change I mentioned above in the ***Debug Results*** section?

**Comment by aciccarello** [comment]
Hello @ovasquez, I haven't looked too much into the issue but I'd be open to a PR. From looking at your **Debug Results** section and the interface, should `IndexSignature` alse use the default case logic?

@shlomiassaf I know you had worked on extracting the serializer code. Do you have any incite into why the logic does not seem to match the interface here?

**Comment by ovasquez** [comment]
@aciccarello you're right, according to the interface all signatures should be given in array form, in that case the switch statement can be removed. . I'll send a PR during the next days. Thanks for the help!

**Comment by aciccarello** [comment]
Resolved in #721
